ts/nni_manager/common/training_service_v3.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ *  TODO
+ *  This type is planned to be an interface.
+ *  To minimize the change, it is currently identical to TrialJobMetric.data
+ **/
+export type Metric = string;
+
+/**
+ *  TODO
+ *  This type is planned to be an interface.
+ *  To minimize the change, it is currently identical to HyperParameters.value
+ **/
+export type Parameter = string;
+
+export interface EnvironmentInfo {
+    id: string;
+}
+
+export interface TrainingServiceV3 {
+    init(): Promise<void>;
+    start(): Promise<EnvironmentInfo[]>;
+    stop(): Promise<void>;
+
+    uploadDirectory(directoryName: string, path: string): Promise<void>;
+    createTrial(environmentId: string, trialCommand: string, directoryName: string): Promise<string | null>;
```

Comment:
Would `createTrial` send a parameter with it?

Reply:
By design, no.
If the trial does not invoke `get_next_parameter()` or `get_current_parameter()`, no parameter will be generated.

In current implementation, yes.
The compat layer caches a parameter on creating trial, and send it to first `onRequestParameter()`.